packages/cdk-routing-split/test/cases/unit/delete.test.ts

```diff
@@ -1,9 +1,47 @@
 import * as when from '../../steps/when';
+import * as given from '../../steps/given';
+import * as then from '../../steps/then';
+import { Key } from '../../../src/delete/interface';
+import { createPK } from '../../../src/shared/make-keys';
 
-describe('When an entity', () => {
-    it('calls delete', async () => {
-        const response = await when.we_invoke_delete();
-        const success = { statusCode: 200, body: true };
-        expect(response).toStrictEqual(success);
+describe('When an api user', () => {
+    let requestBody: any;
+
+    beforeEach(() => {
+        requestBody = given.get_delete_request_body();
+    });
+
+    it('calls delete with invalid key field', async () => {
+        requestBody['key'] = 'invalidKey';
+        const expectedError = {
+            statusCode: 400,
+            body: `Field key is invalid. Valid values are: ${Object.values(Key).join(', ')}`,
+        };
+
+        const response = await when.we_invoke_delete(requestBody);
+
+        expect(response).toStrictEqual(expectedError);
+    });
+
+    it('calls delete with missing value field', async () => {
+        requestBody['value'] = '';
+        const expectedError = { statusCode: 400, body: 'Field value is required.' };
+
+        const response = await when.we_invoke_delete(requestBody);
+
+        expect(response).toStrictEqual(expectedError);
+    });
+
+    it('calls delete with no matching record in table', async () => {
+        requestBody = given.get_delete_request_body();
```

Comment:
This line is redundant, since get_delete_request_body is called before every test.